### PR TITLE
A/B Test – Signed out messaging

### PR DESCRIPTION
**Pull request by sammorrisdesign-zz**
If the user is signed out and not on mobile, we show this little message...

cc @rich-nguyen @phamann 

screen shot 2015-03-12 at 11 22 27

### CODE REVIEW

File: common/app/views/fragments/header.scala.html [hunk 1]
```diff
@@ -20,6 +20,13 @@
                         <span class="js-profile-info control__info" data-test-id="sign-in-name">sign in</span>
                     </a>
                     <div class="js-profile-popup"></div>
+                    <div class="js-popup-signed-out popup popup--signed-out u-h">
+                        <a class="popup--signed-out__link" href="@Configuration.id.url/signin" data-link-name="Signed Out Link"></a>
+                        You are signed out of the Guardian. <span class="popup--signed-out__highlight">Sign In</span>
+                        <button class="js-popup-toggle u-button-reset popup__toggle">
```
Comment (crifmulholland) [inline common/app/views/fragments/header.scala.html]: We need tracking on the close link. 

Can we add "Signed out Link | Close"... We'll need to double-check this works on buttons too.

File: static/src/javascripts/projects/common/modules/experiments/tests/signed-out.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        this.idealOutcome = 'Sign in rate increases';
+
+        this.canRun = function () {
+            if (cookies.get('GU_SO') && !(cookies.get('GU_U')) && (detect.getBreakpoint() !== 'mobile') && !(storage.local.get('gu.ab.signedout'))) {
+                return true;
+            }
+        };
+
+        /**
+         * nothing happens in here, we just use this to bucket users
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/experiments/tests/signed-out.js]: lies! copy and pasted lies!

File: static/src/javascripts/projects/common/modules/experiments/tests/signed-out.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        this.description = 'Testing to see if alerting users on sign out will increase the sign in rate.';
+        this.audience = 1;
+        this.audienceOffset = 0;
+        this.successMeasure = 'Users sign back in after signing out.';
+        this.audienceCriteria = 'All users who have signed out and are not signed in and are not on mobile and have not closed the message before';
+        this.dataLinkNames = 'Signed Out Link';
+        this.idealOutcome = 'Sign in rate increases';
+
+        this.canRun = function () {
+            if (cookies.get('GU_SO') && !(cookies.get('GU_U')) && (detect.getBreakpoint() !== 'mobile') && !(storage.local.get('gu.ab.signedout'))) {
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/experiments/tests/signed-out.js]: I think this is fine, but a subtle issue i've just realised is that `canRun` is really intended as an 'are you a valid candidate' query, such as 'is this mobile'. So does a false from `canRun` mean you're not considered a test participant? Can't remember. I think what you've done is sensible, so let's examine the population size of each variant. We want them to be equal.

**Comment by rich-nguyen** [comment]
:+1: cool

**Comment by marchibbins** [comment]
FYI, this is running on all Identity pages but the component is absent from `identityHeader.scala.html` ([note](https://github.com/guardian/frontend/blob/fd3c7beb4d8d48a75fb805807809bdd62bde5658/common/app/views/fragments/header.scala.html#L6-L9)). So you could exclude by `config.page.section !== 'identity'` in the `canRun`, else expect your data to likely be skewed.

**Comment by sammorrisdesign** [comment]
@marchibbins Don't you have to be signed in to visit the identity pages though?

**Comment by marchibbins** [comment]
@sammorrisdesign most pages but not all, it's everything under the `profile` subdomain, e.g. [user activity](https://profile.theguardian.com/user/id/4453971).

File: static/src/stylesheets/module/nav/_navigation.scss [hunk 4]
```diff
@@ -813,3 +813,59 @@ $burger-width: 16px;
...
+
+/**
+ * Signed Out Popup styling
+ */
+ 
+$signedout-arrow-size: $gs-gutter / 2;
+$signedout-padding: $gs-gutter / 3;
+ 
+.popup.popup--signed-out {
+    left: 0 !important;
```
Comment (OliverJAsh) [inline static/src/stylesheets/module/nav/_navigation.scss]: :scream:

**Pull request merged**